### PR TITLE
Added support for cache dictionary for select method. This allows finer-...

### PR DESCRIPTION
**Pull request by rpwatkins**
Added support for cache dictionary for select method. This allows finer-grained cache invalidation. This matches the changes to the count method in objects.py.

### CODE REVIEW

**Comment by niphlod** [comment]
add tests if you want to change / modify / add something to the existing API.

File: pydal/adapters/base.py [hunk 1]
```diff
@@ -1207,9 +1207,17 @@ def _select_aux(self, sql, fields, attributes):
...
         else:
-            (cache_model, time_expire) = cache
-            key = self.uri + '/' + sql + '/rows'
-            key = hashlib_md5(key).hexdigest()
+            if isinstance(cache, dict):
+                cache_model = cache['model']
+                time_expire = cache['expiration']
+                key = cache.get('key')
+                if not key:
+                    key = self.uri + '/' + sql
```
Comment (ilvalle) [inline pydal/adapters/base.py]: Mind that this concatenation is different from the one we have few lines below

**Comment by ilvalle** [comment]
At this point, we should refactor the code in order to delegate the creation of the key to a single method

**Comment by mdipierro** [comment]
I will take the patch, make the recommended changes but we should eventually follow ilvalle's advice.

**Comment by gi0baro** [comment]
what about the tests, as mentioned by @niphlod ?

**Pull request merged**